tests/benchmark/bloatnet/test_bloatnet.py

```diff
@@ -0,0 +1,242 @@
+"""
+abstract: Optimized BloatNet tests using CREATE2 pre-deployed contracts.
+
+    This test uses CREATE2 factory-deployed contracts at deterministic addresses.
+    Deploy contracts using: python3 scripts/deploy_bloatnet_simple.py --num-contracts 1838
+
+    The CREATE2 pattern allows:
+    - Contracts to be deployed from any account
+    - Reuse across different test scenarios
+    - Deterministic addresses regardless of deployer nonce
+"""
+
+import pytest
+
+from ethereum_test_forks import Fork
+from ethereum_test_tools import (
+    Account,
+    Address,
+    Alloc,
+    Block,
+    BlockchainTestFiller,
+    Bytecode,
+    Transaction,
+    keccak256,
+)
+from ethereum_test_tools.vm.opcode import Opcodes as Op
+
+REFERENCE_SPEC_GIT_PATH = "DUMMY/bloatnet.md"
+REFERENCE_SPEC_VERSION = "1.0"
+
+# Gas cost constants - used to calculate required contracts for any gas limit
+# See README.md for detailed breakdown of these costs
+GAS_PER_CONTRACT_BALANCE_EXTCODESIZE = 2707  # BALANCE(cold) + EXTCODESIZE(warm)
+GAS_PER_CONTRACT_BALANCE_EXTCODECOPY = 5007  # BALANCE(cold) + EXTCODECOPY(warm, 24KB)
+
+# Configuration for CREATE2 pre-deployed contracts
+# These values must match what the deployment script generates
+# UPDATE THESE VALUES after running deploy_bloatnet_simple.py
+FACTORY_ADDRESS = Address("0x07EADb2f6b02Bb9fE994c0AeFf106625c0d3C93f")  # UPDATE THIS
+INIT_CODE_HASH = bytes.fromhex(
+    "8eba8b9d04aea4e6f3d6601ad364021e8e37b6282a0f57e40c635ba7f80aa0cb"
+)  # UPDATE THIS
+NUM_DEPLOYED_CONTRACTS = 1838  # UPDATE THIS - current setup for 5M gas
```

Comment:
Currently using Geth development account which I've used to test that the overall deployment script + fixture generation + execute mode work as expected.

This needs to be updated with the final values after the deployment in Bloatnet (which I'll perform once we all agree this is the way to go).